File: Gateway/src/main/java/com/quizGpt/Gateway/Quiz/Dto/QuestionDto.java
package com.quizGpt.formManagement.Quiz.Dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class QuestionDto {
    // private Long questionId;
    private String text;

    private List<OptionDto> options;
    private List<AnswerDto> answers;
}

File: resource-server/src/quiz/dtos/questionDto.ts
import { AnswerDto } from "./answerDto.js";
import { OptionDto } from "./optionDto.js";
import { QuizDto } from "./quizDto.js";

export class QuestionDto {
  questionId: number;
  text: String;
  options: OptionDto[];
  answers: AnswerDto[];
  quiz: QuizDto;

  constructor(
    questionId: number,
    text: String,
    options: OptionDto[],
    answers: AnswerDto[],
    quiz: QuizDto
  ) {
    this.questionId = questionId;
    this.text = text;
    this.options = options;
    this.answers = answers;
    this.quiz = quiz;
  }
}
